### PR TITLE
Remove `extern crate prusti_contracts` from Prusti

### PR DESCRIPTION
**Pull request by JonasAlaif**

### CODE REVIEW

File: docs/user-guide/src/tour/tour-src/05-chapter-2-3.rs [hunk 1]
```diff
@@ -5,9 +5,8 @@
     To this end, we also need to implement a length function for links
 */
 
-// (1) required to enable Prusti's annotations (implemented as Rust macros/attributes)
-extern crate prusti_contracts;
-use prusti_contracts::*; 
+// (1) simplifies writing Prusti's annotations (implemented as Rust macros/attributes)
```
Comment (Aurel300) [inline docs/user-guide/src/tour/tour-src/05-chapter-2-3.rs:8]: Are you sure about "simplifies"? When we process nested attributes, do we also allow `prusti_contracts::_`? E.g.

```
#[prusti_contracts::requires(...)]
#[prusti_contracts::ensures(...)]
fn foo() {}
```

Would the second attribute parse successfully? (Related #178)

**Comment by Aurel300** [comment]
Thanks, this was also bugging me.

File: docs/user-guide/src/tour/tour-src/05-chapter-2-3.rs (hunk 1 above)
Comment (JonasAlaif) [reply docs/user-guide/src/tour/tour-src/05-chapter-2-3.rs:8]: From what I remember I think things should work, maybe with minor span glitches - it'll just end up running the proc macro twice. Is there actually any reason to handle all of the pres/posts in one go vs. letting each one fire individually? It seems that the latter would be more resilient to this kind of stuff

Comment (JonasAlaif) [reply docs/user-guide/src/tour/tour-src/05-chapter-2-3.rs:8]: A simple fix would be to handle both cases of `#[requires(...)]` and `#[prusti_contracts::requires(...)]`, see latest push (though `"generic"` and `"concrete"` annotations would still throw an error I guess)

**Comment by zgrannan** [comment]
Finished benchmarking commit (c449775827cf7227253197dd9c2563e04953a62e): [comparison url](http://3.94.193.1:2346/compare.html?start=81c84a036c9ae233806706da639be45165593539&end=c449775827cf7227253197dd9c2563e04953a62e&stat=wall-time).

Benchmark returns the average of 3 consecutive runs of the benchmark suite, performed after an initial 'warm-up' run. An external Prusti server is maintained over the 4 runs.
         
### [Wall time](http://3.94.193.1:2346/compare.html?start=81c84a036c9ae233806706da639be45165593539&end=c449775827cf7227253197dd9c2563e04953a62e&stat=wall-time)
This benchmark run did not return any relevant results for this metric.

### [Instruction count (excludes Viper)](http://3.94.193.1:2346/compare.html?start=81c84a036c9ae233806706da639be45165593539&end=c449775827cf7227253197dd9c2563e04953a62e&stat=instructions:u)
This benchmark run did not return any relevant results for this metric.

### [Cycles (excludes Viper)](http://3.94.193.1:2346/compare.html?start=81c84a036c9ae233806706da639be45165593539&end=c449775827cf7227253197dd9c2563e04953a62e&stat=cycles:u)
<details>
<summary>Results</summary>


- Primary benchmarks: 🎉 relevant improvement found
- Secondary benchmarks: no relevant changes found

|            | mean[^1] | max | count[^2] |
|:----------:|:--------:|:---:|:---------:|
| Regressions 😿 <br /> (primary) | N/A      | N/A | 0         |
| Regressions 😿 <br /> (secondary) | N/A      | N/A | 0         |
| Improvements 🎉 <br /> (primary) | -2.0%    | -2.0% | 1         |
| Improvements 🎉 <br /> (secondary) | N/A      | N/A | 0         |
| All 😿🎉 (primary) | -2.0%    | -2.0% | 1         |
</details>

[^1]: *the arithmetic mean of the percent change*
[^2]: *number of relevant changes*

If you disagree with this performance assessment, please file an issue in [prusti-dev/prusti-perf](https://github.com/prusti-dev/prusti-perf/issues/new).

@rustbot label: -perf-regression

**Pull request merged**